Fixes a bug in serializing min/max plus one more.

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35705 Enable relu fusion with prepacked linear/conv.
* **#35850 Fixes a bug in serializing min/max plus one more.**

Summary:
1. Clamping values were not being propagated through all the structures
and hence were not being serialized.
2. Moved to using Scalar for min/max instead of float. Reason being, the
fusion for hardtanh_ does not work. During sub graph rewrite we direct
values from hardtanh_ to preacking ops, but since they expect float
values, the types conflict and we cannot serialize the model.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20807523](https://our.internmc.facebook.com/intern/diff/D20807523)

kimishpatel committed:
Fixes a bug in serializing min/max plus one more.

Summary:
1. Clamping values were not being propagated through all the structures
and hence were not being serialized.
2. Moved to using Scalar for min/max instead of float. Reason being, the
fusion for hardtanh_ does not work. During sub graph rewrite we direct
values from hardtanh_ to preacking ops, but since they expect float
values, the types conflict and we cannot serialize the model.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

dreiss commented on aten/src/ATen/native/xnnpack/OpContext.cpp:
```diff
@@ -11,18 +11,20 @@ c10::intrusive_ptr<LinearOpContext>
               weight,
               bias,
-              output_min ? static_cast<float>(*output_min)
+              output_min ? (*output_min).to<float>()
```
Can you write this as `output_min->to<float>()`?

kimishpatel committed:
Update on "Fixes a bug in serializing min/max plus one more."


Summary:
1. Clamping values were not being propagated through all the structures
and hence were not being serialized.
2. Moved to using Scalar for min/max instead of float. Reason being, the
fusion for hardtanh_ does not work. During sub graph rewrite we direct
values from hardtanh_ to preacking ops, but since they expect float
values, the types conflict and we cannot serialize the model.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20807523](https://our.internmc.facebook.com/intern/diff/D20807523)

[ghstack-poisoned]

kimishpatel committed:
Update on "Fixes a bug in serializing min/max plus one more."


Summary:
1. Clamping values were not being propagated through all the structures
and hence were not being serialized.
2. Moved to using Scalar for min/max instead of float. Reason being, the
fusion for hardtanh_ does not work. During sub graph rewrite we direct
values from hardtanh_ to preacking ops, but since they expect float
values, the types conflict and we cannot serialize the model.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20807523](https://our.internmc.facebook.com/intern/diff/D20807523)

[ghstack-poisoned]

kimishpatel committed:
Update on "Fixes a bug in serializing min/max plus one more."


Summary:
1. Clamping values were not being propagated through all the structures
and hence were not being serialized.
2. Moved to using Scalar for min/max instead of float. Reason being, the
fusion for hardtanh_ does not work. During sub graph rewrite we direct
values from hardtanh_ to preacking ops, but since they expect float
values, the types conflict and we cannot serialize the model.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D20807523](https://our.internmc.facebook.com/intern/diff/D20807523)

[ghstack-poisoned]
